razorpay/hypertrace-ui
# feat: save Explorer queries to localStorage

## Description
Initial PR for implementation of https://github.com/hypertrace/hypertrace-ui/issues/1551

### Testing
- Save Query button only appears when saved-queries feature is enabled.
- Save Query button is disabled if not filters are selected.
- Notification pops up when a query is saved successfully.
- Saved query is available in localStorage.
- Edge case: Saving a query works when no earlier saved preferences are available.

### Checklist:

### Documentation
Not sure if I should update the docs for this feature since we're merging to `rzp_main`.

### Screenshot

<img width="911" alt="Screenshot 2022-05-11 at 12 35 37 PM" src="https://user-images.githubusercontent.com/29686866/167789435-de858f66-dcae-4315-8be1-6c771a113d57.png">

### Errors

```ts
this.notificationService.createSuccessToast('Query Saved Successfully!');
```

This line of code throws the following error in the console, although the notification pops up correctly on the screen.

<img width="1082" alt="Screenshot 2022-05-06 at 8 30 54 AM" src="https://user-images.githubusercontent.com/29686866/167060124-d76de6c2-f3a0-4bc0-9e80-b6e000958ef9.png">

I've created an Issue to discuss this:
https://github.com/hypertrace/hypertrace-ui/issues/1592
